ambry-store/src/test/java/com/github/ambry/store/BlobStoreCompactorTest.java

```diff
@@ -1235,13 +1234,27 @@ public void undeleteSameIndexSegmentTest() throws Exception {
     state.addUndeleteEntry((MockId) p9.getKey());
     state.addUndeleteEntry((MockId) p9.getKey(), (short) 2);
     state.addPutEntries(1, CuratedLogIndexState.PUT_RECORD_SIZE, Utils.Infinite_Time);
-    // Make sure we have two log segments
-    writeDataToMeetRequiredSegmentCount(2, null);
+
+    writeDataToMeetRequiredSegmentCount(1, null);
+    // This is the end of the first log segment
+
+    // log segment 1
+    // Index Segment 1.1 P(expired) D U T -> P(expire) U T
+    long p10Expiration = state.time.milliseconds() + 10000;
+    IndexEntry p10 = state.addPutEntries(1, CuratedLogIndexState.PUT_RECORD_SIZE, p10Expiration).get(0);
+    state.addDeleteEntry((MockId) p10.getKey());
+    state.addUndeleteEntry((MockId) p10.getKey());
+    state.makePermanent((MockId) p10.getKey(), false);
+    state.addPutEntries(1, CuratedLogIndexState.PUT_RECORD_SIZE, Utils.Infinite_Time);
+
+    // Make sure we have 3 log segments
+    writeDataToMeetRequiredSegmentCount(3, null);
     state.reloadIndex(true, false);
 
     String logSegmentName = p1.getValue().getOffset().getName();
-    List<String> segmentsUnderCompaction = Collections.singletonList(logSegmentName);
-    long deleteReferenceTimeMs = state.time.milliseconds() + 10000;
+    List<String> segmentsUnderCompaction = getLogSegments(0, 1);
+    state.advanceTime(p10Expiration - state.time.milliseconds() + 1000);
+    long deleteReferenceTimeMs = state.time.milliseconds();
     long endOffsetOfSegmentBeforeCompaction = state.log.getSegment(logSegmentName).getEndOffset();
     int indexSegmentCountBeforeCompaction = state.index.getIndexSegments().size();
```

Comment:
`indexSegmentCountBeforeCompaction` is never accessed, did you want to verify this count originally?

Reply:
oh I see you removed an assert and make this assigned but never accessed.